### PR TITLE
Fixes #691: Added spacing (padding) setting to text with background paragraph type.

### LINKED ISSUE #691: Add padding options to Text with Background paragraph type
## Motivation:
I want the row to be extra fat or extra skinny around my text.

## Requirements:
- Add ability to adjust padding around text (use more content editor friendly language)

- Apply the following options to the div surrounding the container `<div data-quickedit-entity-id="paragraph/1" class="background-wrapper bg-azurite bg-triangles-centered full-width-background mb-4 paragraph paragraph--type--az-text-background paragraph--view-mode--default py-5 text-with-background" data-quickedit-entity-instance-id="0">`

- Space around content
  - Zero | py-0
  - 1 (0.25rem | ~4px) | py-1
  - 2 (0.5rem | ~8px) | py-2
  - 3 (1.0rem | ~16px) | py-3
  - 4 (1.5rem | ~24px) | py-4
  - 5 (3.0rem | ~48px) | py-5 (default selection)
  - 6 (4.0rem | ~64px) | py-6
  - 7 (5.0rem | ~80px) | py-7
  - 8 (6.0rem | ~96px) | py-8
  - 9 (7.0rem | ~112px) | py-9
  - 10 (8.0rem | ~128px) | py-10

Should adjust this padding:
image

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Adds additional spacing (padding) setting to existing Text with Background paragraph behavior settings.

The database update contained here is still untested and probably needs work.

## Related Issue
#691 

## How Has This Been Tested?

Locally with lando (but not the DB update yet).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by smayden** [comment]
This change will be so useful!

Looking good in my testing, but I notice the content doesn't seem vertically centered (closer to the top):

image

**Comment by joeparsons** [comment]
@danahertzberg FWIW, removing this style from the Quickstart module has no effect since we're adding both the `.text-with-background` and `.background-wrapper` to Text with background paragraphs currently:
```
.text-with-background {
  padding: 3em 0;
}
```

**Comment by danahertzberg** [comment]
> @danahertzberg FWIW, removing this style from the Quickstart module has no effect since we're adding both the `.text-with-background` and `.background-wrapper` to Text with background paragraphs currently:
> 
> ```
> .text-with-background {
>   padding: 3em 0;
> }
> ```

@joeparsons Sounds good. I think if we just update the Arizona Bootstrap documentation to say that you can adjust padding with the utility classes, that will help fill in any gaps.

**Comment by danahertzberg** [comment]
> Looking good in my testing, but I notice the content doesn't seem vertically centered (closer to the top):

@smayden This is because all paragraphs have bottom margin. Adding the `m-0` or `mb-0` class to the paragraph will make the padding even.

**Review by danahertzberg — DISMISSED** [review]

**Comment by joeparsons** [comment]
> @joeparsons Sounds good. I think if we just update the Arizona Bootstrap documentation to say that you can adjust padding with the utility classes, that will help fill in any gaps.

@danahertzberg I was just pointing out that the class that styles that were being provided by the Quickstart module (removed in this PR) were redundant with the Bootstrap class.  Are you suggesting we actually want to remove those styles from `.background-wrapper` in Arizona Bootstrap as well?  Do we need a separate ticket for that?

**Comment by danahertzberg** [comment]
> @danahertzberg I was just pointing out that the class that styles that were being provided by the Quickstart module (removed in this PR) were redundant with the Bootstrap class. Are you suggesting we actually want to remove those styles from `.background-wrapper` in Arizona Bootstrap as well? Do we need a separate ticket for that?

@joeparsons No, sorry for being confusing. I think what you have in this PR is perfect. Only remove the redundant CSS. I would just like to update the bootstrap documentation in the future to share how to adjust padding without Quickstart if desired.

**Comment by joeparsons** [comment]
Add batch sandbox to DB update function to make it be able to handle a large number of existing paragraphs, if necessary.

**Comment by joeparsons** [comment]
I would appreciate review of the database update function included in this PR.

**Comment by danahertzberg** [comment]
Implement Troy's suggestion and remove `text-with-background` class. Then ready to merge after security update.

File: modules/custom/az_paragraphs/az_paragraphs_text_background/az_paragraphs_text_background.install [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  $paragraphs_per_batch = 25;
+
+  $paragraph_ids = \Drupal::entityQuery('paragraph')
+    ->condition('type', 'az_text_background')
+    ->range($sandbox['current'], $sandbox['current'] + $paragraphs_per_batch)
+    ->execute();
+
+  foreach ($paragraph_ids as $id) {
+    $paragraph = Paragraph::load($id);
```
[the remark this answers is not in the document]
Comment (joeparsons) [reply modules/custom/az_paragraphs/az_paragraphs_text_background/az_paragraphs_text_background.install:32]: Added a `!empty` check.

**Review by tadean — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Pull request merged**